libpod/container_internal_linux.go

```diff
@@ -416,9 +417,25 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 
 	// Look up and add groups the user belongs to, if a group wasn't directly specified
 	if !strings.Contains(c.config.User, ":") {
-		for _, gid := range execUser.Sgids {
-			// FIXME: We need to add a flag to containers.conf to not add these for HPC Users.
-			g.AddProcessAdditionalGid(uint32(gid))
+		var haveEnoughGids bool
+
+		if !rootless.IsRootless() {
+			// Running as root, assume there are enough GIDs.
+			haveEnoughGids = true
+		} else {
+			// Check whether the current user namespace has enough gids available.
+			availableGids, err := rootless.GetAvailableGids()
+			if err != nil {
+				return nil, errors.Wrapf(err, "cannot read number of available GIDs")
+			}
+			// Do not check what values, as the user can specify an arbitrary user namespace
+			// to accomodate the requested gids.
+			haveEnoughGids = availableGids > int64(len(execUser.Sgids))
```

Comment:
I don't understand this?  
Don't we want to check each GID in the execUser.Sgids > avialbaleGids?

Reply:
I tried to avoid looking into the user namespace mapping, but I agree the current check doesn't make much sense.

I've changed it to look at each gid and see if it is present in the namespace (current or configured for the container)